Fix(ssrm): expand all example fix

sdwvit opened the pull request.

Commit:
fix example

Signed-off-by: Victor Musienko <victor.m@noibu.com>

Commit:
remove ids

Signed-off-by: Victor Musienko <victor.m@noibu.com>

sdwvit commented on documentation/ag-grid-docs/src/content/docs/server-side-model-grouping/_examples/expand-all-affects-all-rows/main.ts:
```diff
@@ -1,11 +1,5 @@
-import {
-    type GridApi,
```
ng transformer doesn't understand this syntax

sdwvit commented on documentation/ag-grid-docs/src/content/docs/server-side-model-grouping/_examples/expand-all-affects-all-rows/main.ts:
```diff
@@ -28,7 +22,6 @@ const gridOptions: GridOptions = {
         { field: 'silver', aggFunc: 'sum' },
         { field: 'bronze', aggFunc: 'sum' },
     ],
-    ssrmExpandAllAffectsAllRows: (document.querySelector('#input-display-type') as HTMLSelectElement).value === 'true',
```
direct dom manipulations break vue transformer

sdwvit commented on documentation/ag-grid-docs/src/content/docs/server-side-model-grouping/_examples/expand-all-affects-all-rows/index.html:
```diff
@@ -1,11 +1,8 @@
-            ssrmExpandAllAffectsAllRows:
-            <input type="checkbox" id="input-display-type" onchange="onOptionChange()" />
-        </label>
+        <button onclick="onExpandAll(false)">Expand visible rows</button>
```
can't use onExpandAll() as react tranformer inserts an event parameter there which is coerced to truthy value

sdwvit commented on documentation/ag-grid-docs/src/content/docs/server-side-model-grouping/_examples/expand-all-affects-all-rows/main.ts:
```diff
@@ -56,34 +49,34 @@ function getServerSideDatasource(server: any): IServerSideDatasource {
-const gridDiv = document.querySelector<HTMLElement>('#myGrid')!;
-gridApi = createGrid(gridDiv, gridOptions);
+// setup the grid after the page has finished loading
+document.addEventListener('DOMContentLoaded', function () {
```
can't do direct fetch, as dom may not be ready in some frameworks at this point of time

peterjrreynolds commented on documentation/ag-grid-docs/src/content/docs/server-side-model-grouping/_examples/expand-all-affects-all-rows/index.html:
```diff
@@ -1,11 +1,8 @@
-        <button id="collapse" onclick="onCollapseAll()">Collapse rows</button>
-        <label>
-            ssrmExpandAllAffectsAllRows:
-            <input type="checkbox" id="input-display-type" onchange="onOptionChange()" />
```
Can we put this back please (e.g. checkbox plus two buttons) so that the behaviour is consistent with other examples. Look at an example like `omit-function-name-in-header` for details on how to get checkboxes working (grid option value can be set directly to `false`)

peterjrreynolds commented on documentation/ag-grid-docs/src/content/docs/server-side-model-grouping/index.mdoc:
```diff
@@ -131,14 +131,12 @@ The example above demonstrates the following configuration:
-
-// Checking the checkbox applies the following:
+// Second button click expands all group rows, even those not yet loaded:
 api.setGridOptions({ ssrmExpandAllAffectsAllRows: true });
```
Docs pages should never reference `setGridOption` or `updateGridOptions` as these are primarily used by vanilla JS only (`setGridOptions` isn't actually an API method). Frameworks will update properties bound to the grid component.

Should remove this code snippet completely and replace it with a series of bullet points, similar to how we describe other examples.

sdwvit committed:
Update server-side grouping docs to reflect ssrmExpandAllAffectsAllRows configuration

- Added checkbox in example to demonstrate ssrmExpandAllAffectsAllRows configuration
- Updated description to explain how expand/collapse operations interact with this flag
- Linked example code to show configuration effect on row expansion behavior
- Cleaned up redundant text in documentation while preserving technical accuracy
